Source/Urho3D/Core/ProcessUtils.cpp

```diff
@@ -412,6 +423,63 @@ ea::string GetPlatformName()
     }
 }
 
+unsigned GetNumPhysicalCPUs()
+{
+    int cpuCount = 0;
+#if defined(IOS)
+    host_basic_info_data_t data;
+    mach_msg_type_number_t infoCount;
+    infoCount = HOST_BASIC_INFO_COUNT;
+    host_info(mach_host_self(), HOST_BASIC_INFO, (host_info_t)&data, &infoCount);
+    cpuCount = data.physical_cpu;
+#elif defined(TVOS)
+    cpuCount = SDL_TVOS_GetActiveProcessorCount();
+#elif defined(APPLE) || defined(__FreeBSD__)
+    size_t size = sizeof(cpuCount);
+    sysctlbyname("hw.physicalcpu", &cpuCount, &size, nullptr, 0);
+#elif defined(__linux__)
+    FILE* fp = fopen("/proc/cpuinfo", "rb");
+    if (fp)
+    {
+        char line[1024];
+        while (fgets(line, sizeof(line), fp))
+        {
+            int id = 0;
+            if (sscanf(line, "core id%*[\t]: %d", &id) == 1)
+                cpuCount = Max(cpuCount, static_cast<unsigned>(id + 1));
+        }
+        fclose(fp);
+    }
+#elif defined(__EMSCRIPTEN__)
+#ifdef __EMSCRIPTEN_PTHREADS__
```

Comment:
How is that better than `std::thread::hardware_concurrency`?
Also, let's use macros from here whenever possible (I don't know if it's possible in this specific case):
https://github.com/rbfx/rbfx/blob/master/Source/Urho3D/Urho3D.h#L113

Reply:
Right, its not better. I just restored old code 😁